record_generator.go

```diff
@@ -26,15 +26,33 @@ import (
 	"github.com/google/uuid"
 )
 
-type recordGenerator interface {
-	generate() (sdk.Record, error)
+type recordGenerator struct {
+	config RecordConfig
+	cached sdk.RawData
 }
 
-type defaultRecordGen struct {
-	config RecordConfig
+func newRecordGenerator(config RecordConfig) recordGenerator {
```

Comment:
Nitpick: If `recordGenerator` is supposed to be used as a pointer (I see all methods are defined on the pointer) we should also return a pointer here.
```suggestion
func newRecordGenerator(config RecordConfig) *recordGenerator {
```

Reply:
Done.